src/java/org/apache/cassandra/utils/SyncUtil.java

```diff
@@ -105,62 +63,19 @@ public static MappedByteBuffer force(MappedByteBuffer buf)
         Preconditions.checkNotNull(buf);
         if (SKIP_SYNC)
         {
-            Object fd = null;
-            try
-            {
-                if (mbbFDField != null)
-                {
-                    fd = mbbFDField.get(buf);
-                }
-            }
-            catch (Exception e)
-            {
-                throw new RuntimeException(e);
-            }
-            //This is what MappedByteBuffer.force() throws if a you call force() on an umapped buffer
-            if (mbbFDField != null && fd == null)
-                throw new UnsupportedOperationException();
             return buf;
         }
         else
         {
             return buf.force();
         }
     }
-
+    
     public static void sync(FileDescriptor fd) throws SyncFailedException
     {
         Preconditions.checkNotNull(fd);
-        if (SKIP_SYNC)
-        {
-            boolean closed = false;
-            try
-            {
-                if (fdClosedField != null)
-                    closed = fdClosedField.getBoolean(fd);
-            }
-            catch (Exception e)
-            {
-                throw new RuntimeException(e);
-            }
-
-            int useCount = 1;
-            try
-            {
-                if (fdUseCountField != null)
-                    useCount = ((AtomicInteger)fdUseCountField.get(fd)).get();
-            }
-            catch (Exception e)
-            {
-                throw new RuntimeException(e);
-            }
-            if (closed || !fd.valid() || useCount < 0)
-                throw new SyncFailedException("Closed " + closed + " valid " + fd.valid() + " useCount " + useCount);
```

Comment:
@jacek-lewandowski I was double-checking the patch, shouldn't we actually still do:
```
if (!fd.valid())
                throw new SyncFailedException("Valid " + fd.valid());
```

Reply:
Why calling the method which is supposed to do nothing, should ever end with an exception?

When skip-sync=true, the expectation is that this method is no-op.

Reply:
> When skip-sync=true, the expectation is that this method is no-op.

I am afraid this is not exactly the case.
From the original ticket where `SyncUtil `was added - https://issues.apache.org/jira/browse/CASSANDRA-9403
```
Unit tests aren't testing crash safety so it is a pointless exercise.

Instead we could intercept calls to sync files and check whether it looks like the sync would succeed. Check that the things are not null, mapped, closed etc. Outside of units tests it can go straight to the regular sync call.
```
But I see now we are accounting for the nulls with preconditions already and there is no call for valid() we missed to cover from the methods we override. Moving to commit